pylint/utils.py

```diff
@@ -730,32 +732,93 @@ def register_messages(self, checker):
-                raise InvalidMessageError(
-                    'Message symbol %r is already defined' % msg.symbol)
-            if chkid is not None and chkid != msg.msgid[1:3]:
-                raise InvalidMessageError(
-                    "Inconsistent checker part in message id %r (expected 'x%sxx')"
-                    % (msgid, chkid))
+            self._check_message(msg, chkid)
             chkid = msg.msgid[1:3]
             self._messages[msg.symbol] = msg
-            self._register_alternative_name(msg, msg.msgid)
+            self._register_alternative_name(msg, msg.msgid, msg.symbol)
             for old_id, old_symbol in msg.old_names:
-                self._register_alternative_name(msg, old_id)
-                self._register_alternative_name(msg, old_symbol)
+                self._register_alternative_name(msg, old_id, old_symbol)
             self._msgs_by_category[msg.msgid[0]].append(msg.msgid)
 
-    def _register_alternative_name(self, msg, name):
+    def _register_alternative_name(self, msg, msgid, symbol):
         """helper for register_message()"""
-        if name in self._messages and self._messages[name] != msg:
-            raise InvalidMessageError(
-                'Message symbol %r is already defined' % name)
-        if name in self._alternative_names and self._alternative_names[name] != msg:
-            raise InvalidMessageError(
-                'Message %s %r is already defined' % (
-                    'id' if len(name) == 5 and name[0] in MSG_TYPES else 'alternate name',
-                    name))
-        self._alternative_names[name] = msg
+        self._check_id_and_symbol_consistency(msgid, symbol)
+        self._alternative_names[msgid] = msg
+        self._alternative_names[symbol] = msg
+
+    def _check_symbol(self, msgid, symbol):
+        """ Check that a symbol is not already used. """
+        if symbol in self._messages:
+            other_msgid = self._messages[symbol].msgid
+            self._raise_duplicate_msg_id(symbol, msgid, other_msgid)
+        if symbol in self._alternative_names:
+            msg = self._alternative_names[symbol]
+            if msg.symbol == symbol:
+                other_msgid = msg.msgid
+            else:
+                for old_msgid, old_symbol in msg.old_names:
+                    if old_symbol == symbol:
+                        other_msgid = old_msgid
+                        break
+            if msgid != other_msgid:
+                self._raise_duplicate_msg_id(symbol, msgid, other_msgid)
+
+    def _check_msgid(self, msgid, symbol):
+        for m_symbol in self._messages:
+            msg = self._messages[m_symbol]
+            if msg.msgid == msgid:
+                self._raise_duplicate_symbol(msgid, symbol, msg.symbol)
+
+    def _check_message(self, msg, check_id=None, is_old_name=False):
+        """A new messages is correct according to a dict of messages. """
```

Comment:
It's not super obvious what this function does just from reading the docstring, from where does the correctness entail given a dict of messages, what property that dict has etc. I say we should provide some more verbiage to this docstring.

Reply:
I had to guess a lot of thing regarding this dict myself, so I agree. Once I'm able understand or refactor it I'll detail what is in it.